### PR TITLE
feat Invite button

### PR DESCRIPTION
**Pull request by iccir919**

### CODE REVIEW

File: server/routes/add_invite.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+let db = require('../config');
+
+module.exports = function(req, res) {
+	let user = parseInt(req.user.user_id);
```
Comment (jchen54) [inline server/routes/add_invite.js]: ParseInt might not be needed if db is established efficiently

File: server/schema.js [hunk 2]
```diff
@@ -46,5 +46,11 @@ module.exports = (db) => {
     return db.query('CREATE TABLE IF NOT EXISTS friends (\
       user1 BIGINT not null,\
       user2 BIGINT not null);')
+  })
+  .then(() => {
+    return db.query('CREATE TABLE IF NOT EXISTS invites (\
```
Comment (jchen54) [inline server/schema.js]: Should contain a primary key as an identifier

File: src/client/app/InviteButton.jsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	getUsersInvites(){
+		$.ajax({
+			url: '/invites',
+			method: 'GET',
+			contentType: 'application/json',
+			success: function(data){
+				this.setState({
+					invites: data
+				})
+			}.bind(this),
```
Comment (jchen54) [inline src/client/app/InviteButton.jsx]: Use arrow function to avoid binding inside AJAX

File: src/client/app/InviteButton.jsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+		}
+		$.ajax({
+			url: '/invites',
+			method: 'POST',
+			data: JSON.stringify(inviteData),
+			contentType: 'application/json',
+	    success: function() {
+	    	this.getUsersInvites();
+
+	    }.bind(this),
```
Comment (jchen54) [inline src/client/app/InviteButton.jsx]: Use arrow function to avoid binding inside AJAX

File: src/client/app/InviteButton.jsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+	componentWillMount(){
+		this.getUsersInvites();
+	}
+
+	getUsersInvites(){
+		$.ajax({
+			url: '/invites',
+			method: 'GET',
+			contentType: 'application/json',
```
Comment (jchen54) [inline src/client/app/InviteButton.jsx]: GET requests don't require contentType

File: src/client/app/InviteButton.jsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+      			<div className="row">
+      				<div className="col-md-12">
+      					<h4 className='create'>{this.state.invites ? this.state.invites.length : null} Previously Invited:</h4>
+      					<InviteList invites={this.state.invites}> </InviteList>
+	      				<h3 className='create'>Input name and email:</h3>
+	              <h4 className='create'>Name:</h4>
+	              <input 
+	                value={this.state.title} 
+	                type="text"
+	                onChange={this.handleChange.bind(this, 'name')} required
```
Comment (jchen54) [inline src/client/app/InviteButton.jsx]: Unnecessary binding: handleChange is already binded on line 17

File: src/client/app/InviteButton.jsx [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+	              />
+	              <h4 className='create'>Email: </h4>
+	              <input 
+	                value={this.state.title} 
+	                type="text"
+	                onChange={this.handleChange.bind(this, 'email')} required
```
Comment (jchen54) [inline src/client/app/InviteButton.jsx]: Unnecessary binding: handleChange is already binded on line 17

File: src/client/app/InviteButton.jsx [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+		this.setState(newState);
+	}
+
+  render(){
+    return(
+      <div>
+      <div><button id="invite" onClick={this.showModal.bind(this)} className="col-md-4 col-md-offset-4">Send Invite To App</button></div>
+      <Modal ref="modal">
+      	<div className="container-fluid">
+      		<form onSubmit={this.handleSubmit.bind(this)}>
```
Comment (jchen54) [inline src/client/app/InviteButton.jsx]: Unnecessary binding: handleSubmit is already binded on line 16

File: server/routes/add_invite.js (hunk 1 above)
Comment (kay-lee) [inline server/routes/add_invite.js]: - why was it necessary to have parseInt again?

File: src/client/app/InviteButton.jsx (hunk 5 above)
Comment (misteryeo) [inline src/client/app/InviteButton.jsx]: Don't believe this is necessary for a GET request

File: server/routes/get_events.js [hunk 9]
```diff
@@ -6,7 +6,6 @@ let model = require('../models/index.js');
 //query database to get all events
 
 module.exports = (req, res) => {
-  console.log(req.user.user_id);
```
Comment (kay-lee) [inline server/routes/get_events.js]: - thanks for figuring out that this file had console.log that logs id :)

File: server/schema.js [hunk 10]
```diff
@@ -46,5 +46,11 @@ module.exports = (db) => {
...
+      user_id BIGINT not null,\
```
Comment (misteryeo) [inline server/schema.js]: Still wondering if a join table would be more appropriate to adjust for multiple users inviting multiple friends

File: src/client/app/InviteButton.jsx [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+	              <h4 className='create'>Name:</h4>
+	              <input 
+	                value={this.state.title} 
+	                type="text"
+	                onChange={this.handleChange.bind(this, 'name')} required
+	              />
+	              <h4 className='create'>Email: </h4>
+	              <input 
+	                value={this.state.title} 
+	                type="text"
```
Comment (jchen54) [inline src/client/app/InviteButton.jsx]: Should use type="email" to validate email format

File: server/schema.js (hunk 10 above)
Comment (kay-lee) [inline server/schema.js]: - from MySQL, I found BIGINT has the biggest storage (https://dev.mysql.com/doc/refman/5.7/en/integer-types.html) 
- would like to know why you specifically used BIGINT!

File: server/server.js [hunk 12]
```diff
@@ -43,6 +43,14 @@ app.post('/delete/owner', passport.authenticate('facebook-token'), handler.delet
 
 app.post('/checkStatus', passport.authenticate('facebook-token'), handler.checkStatus);
 
+app.get('/invites', passport.authenticate('facebook-token'), handler.getInvites);
+
+app.post('/invites', passport.authenticate('facebook-token'), handler.addInvite);
```
Comment (kay-lee) [inline server/server.js]: - what does this part "passport.authenticate('facebook-token')" do here? (clarification)

File: src/client/app/Homepage.jsx [hunk 13]
```diff
@@ -50,11 +51,13 @@ class Homepage extends React.Component {
            <h2 id='userName'>Welcome <span id="headerName">{this.props.userName}</span></h2>
             <LogoutButton />
           </div>
+          <InviteButton />
           <CreateEventButton
           history={this.props.history}
           friends={this.props.friends}
           getEvents={this.props.getEvents}/>
-          <br /><br />
+          <br /><br /><br /><br />
```
Comment (kay-lee) [inline src/client/app/Homepage.jsx]: - does having multiple <br />s actually give more space?

File: src/client/app/InviteButton.jsx [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+			contentType: 'application/json',
+			success: function(data){
+				this.setState({
+					invites: data
+				})
+			}.bind(this),
+	    error: function(err) {
+	      console.log('unsuccessful  request to get all invites', err);
+	    }	
+		});
```
Comment (kay-lee) [inline src/client/app/InviteButton.jsx]: - can we please move this "});" between the previous and next }s?

File: src/client/app/InviteList.jsx [hunk 15]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+var InviteList = (props) => (
+	<div className="invite-list">
+	{props.invites.map((invite) =>
+		<div>Name: <b>{invite.name}</b>            Email: <b>{invite.email} </b></div>
```
Comment (jchen54) [inline src/client/app/InviteList.jsx]: Should use proper unordered list and list entries with unique key identifier

File: src/client/app/InviteButton.jsx [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+	                onChange={this.handleChange.bind(this, 'email')} required
+	              />	              
+      				</div>
+      				<div>
+      					<br></br>
+      				</div>
+              <div className="col-md-12">
+              	<div class="row">
+				            <button type="submit" id="send_invite" >Invite</button>
+				            <button type="button" onClick={this.hideModal.bind(this)} id="exit_invite" className="pull-right" >Done</button>
```
Comment (kay-lee) [inline src/client/app/InviteButton.jsx]: - is there a reason why we have to specify type="button" for <button>?

File: src/client/app/InviteList.jsx (hunk 15 above)
Comment (kay-lee) [inline src/client/app/InviteList.jsx]: - this is cool. I should've done this in my code. Awesome!

File: src/client/app/InviteButton.jsx [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+      	<div className="container-fluid">
+      		<form onSubmit={this.handleSubmit.bind(this)}>
+      			<div className="row">
+      				<div className="col-md-12">
+      					<h4 className='create'>{this.state.invites ? this.state.invites.length : null} Previously Invited:</h4>
+      					<InviteList invites={this.state.invites}> </InviteList>
+	      				<h3 className='create'>Input name and email:</h3>
+	              <h4 className='create'>Name:</h4>
+	              <input 
+	                value={this.state.title}
```
Comment (iccir919) [inline src/client/app/InviteButton.jsx]: delete

File: src/client/app/InviteButton.jsx (hunk 16 above)
Comment (iccir919) [inline src/client/app/InviteButton.jsx]: no

**Pull request closed without merging**